homeassistant/components/rainmachine/update.py

```diff
@@ -99,4 +99,11 @@ def update_from_latest_data(self) -> None:
             UpdateStates.UPGRADING,
             UpdateStates.REBOOT,
         )
-        self._attr_latest_version = data["packageDetails"]["newVersion"]
+
+        # The RainMachine API docs say that multiple "packages" can be updated, but
+        # don't give details on what types exist (which makes it impossible to have
+        # update entities per update type); so, we use the first one (with the idea that
+        # after it succeeds, the entity will show the next update):
+        package_details = data["packageDetails"][0]
+        self._attr_latest_version = package_details["newVersion"]
+        self._attr_title = package_details["packageName"]
```

Comment:
I don't know if this is the right approach long term. It might be better to dynamically create update entities as we learn about them from the data.

Reply:
Actually, I don't think multiple update entities makes sense, since we only get one API call to apply all pending updates.

So, I think this is a reasonable approach.